### ISSUE TITLE
[Bug]: OverflowMenu types not exported despite being in changelog

### ISSUE THREAD

**Issue by patrikhermansson** [carbon-design-system/carbon#16493]
### Package

@carbon/react

### Browser

Chrome, Safari, Firefox, Edge

### Package version

v11.57.0

### React version

v18

### Description

The v11.57.0 changelog says that OverflowMenu now has typescript types (https://github.com/carbon-design-system/carbon/pull/16134). However they are not exported in the actual module. The component is not faulty in any other way and works like before, just missing types.

### Reproduction/example

None

### Steps to reproduce

Use in editor with typescript support 

### Suggested Severity

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by Gururajj77** [comment]
Hey There, as of #16494 , we have exported the `OverflowMenuProps`, you can access them now. Please let me know if you are still not seeing it

**Comment by patrikhermansson** [comment]
I see! It's useful but doesn't work as you'd expect compared to other components in this library, the OverflowMenu component itself is exported now but is untyped. So you can use any props on it and it'll pass typecheck.

**Comment by adamalston** [comment]
https://github.com/carbon-design-system/carbon/issues/16493#issuecomment-2127793190:

> [Y]ou can use any props on it and it'll pass typecheck.

The code I have won't pass the type check. Here's a reproduction: https://stackblitz.com/edit/github-w5m38jfc?file=src%2FApp.tsx. See the type error for `event`.

**Comment by patrikhermansson** [comment]
> The code I have won't pass the type check. Here's a reproduction: https://stackblitz.com/edit/github-w5m38jfc?file=src%2FApp.tsx. See the type error for `event`.

No, the error for event is because the typescript config does not allow implicit any. It's not related to the types on the OverflowMenu. Try changing or adding any random properties to OverflowMenu in your example and you will see that they don't show up as an error. 

<img width="313" alt="Image" src="https://github.com/user-attachments/assets/97ae71c3-4eb6-4ffa-a2fc-2bd91bd86175" />

That's because OverflowMenu has props set to any:

<img width="283" alt="Image" src="https://github.com/user-attachments/assets/938bd768-4b1a-499c-9a4e-6f162889c1e4" />

**Comment by adamalston** [comment]
> No, the error for event is because the typescript config does not allow implicit any. It's not related to the types on the OverflowMenu.

It is related to the types, or rather the lack thereof, for the `OverflowMenu`. If `onClick` was properly typed either directly or indirectly, that error wouldn't show up. See the updated reproduction. There is no type error for `event` in the `Button`'s `onClick` because the prop is typed indirectly via `React.ButtonHTMLAttributes<HTMLButtonElement>`.

**Comment by adamalston** [comment]
https://github.com/carbon-design-system/carbon/pull/18690 changed the types for `OverflowMenu`. Can you check if you're able to use them now?
